python/lsst/validate/base/measurement.py

```diff
@@ -1,46 +1,74 @@
 # See COPYRIGHT file at the top of the source tree.
 from __future__ import print_function, division
-from future.utils import with_metaclass
 
-import abc
 import uuid
 
 import astropy.units as u
-from .datummixin import DatumAttributeMixin
 from .jsonmixin import JsonSerializationMixin
-from .blob import BlobBase, DeserializedBlob
+from .blob import DeserializedBlob
 from .datum import Datum, QuantityAttributeMixin
 from .metric import Metric
 
 
-__all__ = ['MeasurementBase', 'DeserializedMeasurement']
+__all__ = ['MeasurementSet', 'Measurement']
 
 
-class MeasurementBase(with_metaclass(abc.ABCMeta,
-                                     type('NewBase',
-                                          (QuantityAttributeMixin,
-                                           JsonSerializationMixin,
-                                           DatumAttributeMixin),
-                                          {}))):
-    """Base class for Measurement classes.
+class MeasurementSet(object):
+    """A collection of Measurements of Metrics, associated with a MetricSet.
 
-    This class isn't instantiated directly. Instead, developers should
-    subclass `MeasurementBase` to create measurement classes for each
-    metric being measured.
+    Parameters
+    ----------
+    name : `str`
+        The name of this `MetricSet` (usually the name of a package).
+    measurements : `dict` of `str`: `astropy.Quantity`
+        The measurements (astropy.Quantities) of each metric, keyed on the
+        metric name, to be looked up in metric_set.
+    job : `Job`, optional
+        The Job that produced these `Measurements`, linking them to their
+        provenance and other metadata.
+    metric_set : MetricSet, optional
+        A `MetricSet` to extract the metric definitions from. If None, use
+        name from the validate_metrics package.
+    """
+    name = None
+    """`str` the name of the `MetricSet` these `Measurement`s are of."""
+
+    measurements = None
+    """`dict` of all `Measurement` names to `Measurement`s."""
+
+    job = None
+    """`Job` that this MeasurementSet was produced by."""
 
-    Subclasses must (at least) implement the following attributes:
+    def __init__(self, name, measurements, job=None, metric_set=None):
```

Comment:
Like in [my comment](https://github.com/lsst/validate_base/pull/5/files#r106524171) about `MetricSet`, I'm wondering if the `measurements` argument here could just be a list, and then in `__init__` we get the fully-qualified metric name from the measurement objects to build `self.measurements` as a list.